common/src/main/java/org/apache/celeborn/common/network/sasl/SecretRegistryImpl.java

```diff
@@ -33,10 +33,12 @@ public static SecretRegistryImpl getInstance() {
 
   private final ConcurrentHashMap<String, String> secrets = new ConcurrentHashMap<>();
 
+  @Override
   public void register(String appId, String secret) {
     secrets.put(appId, secret);
```

Comment:
put only if absent - and throw exception if already present: in case of race conditions.

Reply:
Btw, this will need to be persisted into ratis, right ? Is that in a follow up PR ?

Reply:
Yes, was planning to add that in a follow up PR.

Reply:
> Btw, this will need to be persisted into ratis, right ? Is that in a follow up PR ?

If Master leader changes, a new client to the new leader will be created, will `RegistrationClientBootstrap` be reused for the new client? If not, seems there will be re-registration error.

Reply:
Once the application has registered, it should not register again even if a new `TransportClient` is created. The `RegistrationClientBootstrap` will be added as a client bootstrap but it will just authenticate the connection and not register.  The below condition would handle that:
```
    if (registrationInfo.getRegistrationState() == RegistrationInfo.RegistrationState.REGISTERED) {
      LOG.info("client has already registered, skip register.");
      doSaslBootstrap(client, channel);
      return;
    }
   ```

Reply:
> Once the application has registered, it should not register again even if a new `TransportClient` is created. The `RegistrationClientBootstrap` will be added as a client bootstrap but it will just authenticate the connection and not register. The below condition would handle that:
> 
> ```
>     if (registrationInfo.getRegistrationState() == RegistrationInfo.RegistrationState.REGISTERED) {
>       LOG.info("client has already registered, skip register.");
>       doSaslBootstrap(client, channel);
>       return;
>     }
> ```

Got it,  thanks for the explanation.